### ISSUE TITLE
pygmt.config() does not set global configuration variables

### ISSUE THREAD

**Issue by MarkWieczorek** [GenericMappingTools/pygmt#1465]
**Description of the problem**
I would like to use pygmt to set a bunch of global configuration variables. This would be done at the beginning of a script and would apply to all new figures.

The problem is that when you create a new figure, the previously set configuration variables appear to be reset to their default values, as shown here:

```
import pygmt

pygmt.config(MAP_FRAME_PEN="blue",
             MAP_TICK_PEN="blue",
             FONT_ANNOT_PRIMARY="blue",
             FONT_LABEL="blue")
fig = pygmt.Figure()
fig.basemap(region=[0, 10, 0, 10],
            projection="X15c/15c",
            frame=["SNWE", "yaf+ly", "xaf+lx"])
fig.show(method='external')
```
![test](https://user-images.githubusercontent.com/8697285/130817493-75403c88-eb07-4b04-9a28-076f5650f6bf.png)

However, if I create the figure first, and then set the config variables, everything works as expected
```
import pygmt

fig = pygmt.Figure()
pygmt.config(MAP_FRAME_PEN="blue",
             MAP_TICK_PEN="blue",
             FONT_ANNOT_PRIMARY="blue",
             FONT_LABEL="blue")
fig.basemap(region=[0, 10, 0, 10],
            projection="X15c/15c",
            frame=["SNWE", "yaf+ly", "xaf+lx"])
fig.show(method='external')
```
![test2](https://user-images.githubusercontent.com/8697285/130817855-45229bc9-230b-447f-92f1-ea70c8d272f7.png)

If this behavior is in fact by design, the question is: How can I change the default configuration values ?

**System information**

Please paste the output of `python -c "import pygmt; pygmt.show_versions()"`:

```
PyGMT information:
  version: v0.3.1
System information:
  python: 3.9.6 | packaged by conda-forge | (default, Jul 11 2021, 03:36:15)  [Clang 11.1.0 ]
  executable: /opt/miniconda3/envs/py39b/bin/python
  machine: macOS-11.5.2-x86_64-i386-64bit
Dependency information:
  numpy: 1.21.2
  pandas: 1.3.2
  xarray: 0.18.2
  netCDF4: 1.5.6
  packaging: 21.0
  ghostscript: 9.54.0
  gmt: 6.1.1
GMT library information:
  binary dir: /opt/miniconda3/envs/py39b/bin
  cores: 8
  grid layout: rows
  library path: /opt/miniconda3/envs/py39b/lib/libgmt.dylib
  padding: 2
  plugin dir: /opt/miniconda3/envs/py39b/lib/gmt/plugins
  share dir: /opt/miniconda3/envs/py39b/share/gmt
  version: 6.1.1
```

**Comment by seisman** [comment]
Your first example works well for me (blue frame, annotations, labels and ticks). Perhaps you should try to bump your GMT and PyGMT versions to the latest?

**Comment by MarkWieczorek** [comment]
It seems to work on 0.4.1. I looked at the release notes before posting this, but didn't see this noted there...

**Comment by seisman** [comment]
I don't think we made any changes to `pygmt.config` recently. Maybe a GMT bug which was fixed in GMT 6.2.0.
